src/components/widgets/filesystem/FileSystem.vue

```diff
@@ -437,13 +437,18 @@ export default class FileSystem extends Mixins(StateMixin, FilesMixin, ServicesM
         // Clear selected bulk items if we're navigating folders.
         this.selected = []
       } else {
-        // Open the context menu
-        this.contextMenuState.x = e.clientX
-        this.contextMenuState.y = e.clientY
-        this.contextMenuState.file = item
-        this.$nextTick(() => {
-          this.contextMenuState.open = true
-        })
+        if (e.type === 'click' && ['txt', 'md', 'conf', 'cfg', 'ini', 'sh'].includes((item as AppFile).extension)) {
```

Comment:
Please don't use `item as AppFile`!

TypeScript is a lot smarter than JavaScript and has a fantastic feature called [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html)!

`AppFile` has been defined as having `type: 'file'`, so all we need here is to check the `type` to make item be of `AppFile`.

Given your code above, just replace it with this:

```typescript
if (item.type === 'file' && e.type === 'click' && ['txt', 'md', 'conf', 'cfg', 'ini', 'sh'].includes(item.extension)) {
  this.handleFileOpenDialog(item)
// rest of stuff
```